app/ide-desktop/lib/dashboard/mock/authentication/src/authentication/cognito.ts

```diff
@@ -0,0 +1,775 @@
+    },
+    forgotPassword: {
+        userNotFound: 'Username not found. Please register first.',
+        userNotConfirmed: `Cannot reset password for user with an unverified email. \
+Please verify your email first.`,
+    },
+}
+
+// ================
+// === UserInfo ===
+// ================
+
+// The names come from a third-party API and cannot be changed.
+/* eslint-disable @typescript-eslint/naming-convention */
+/** Attributes returned from {@link amplify.Auth.currentUserInfo}. */
+interface UserAttributes {
+    email: string
+    email_verified: boolean
+    sub: string
+    'custom:fromDesktop'?: string
+    'custom:organizationId'?: string
+}
+/* eslint-enable @typescript-eslint/naming-convention */
+
+// ====================
+// === AmplifyError ===
+// ====================
+
+/** Error thrown by the AWS Amplify library when an Amplify error occurs.
+ *
+ * Some Amplify errors (e.g., network connectivity errors) can not be resolved within the
+ * application. Un-resolvable errors are allowed to flow up to the top-level error handler. Errors
+ * that can be resolved must be caught and handled as early as possible. The {@link KNOWN_ERRORS}
+ * map lists the Amplify errors that we want to catch and convert to typed responses.
+ *
+ * # Handling Amplify Errors
+ *
+ * Use the {@link isAmplifyError} function to check if an `unknown` error is an
+ * {@link AmplifyError}. If it is, use the {@link intoAmplifyErrorOrThrow} function to convert it
+ * from `unknown` to a typed object. Then, use the {@link KNOWN_ERRORS} to see if the error is one
+ * that must be handled by the application (i.e., it is an error that is relevant to our business
+ * logic). */
+interface AmplifyError extends Error {
+    /** Error code for disambiguating the error. */
+    code: string
+}
+
+/** Hint to TypeScript if we can safely cast an `unknown` error to an {@link AmplifyError}. */
+function isAmplifyError(error: unknown): error is AmplifyError {
+    if (error != null && typeof error === 'object') {
+        return 'code' in error && 'message' in error && 'name' in error
+    } else {
+        return false
+    }
+}
+
+/** Convert the `unknown` error into an {@link AmplifyError} and returns it, or re-throws it if
+ * conversion is not possible.
+ * @throws If the error is not an amplify error. */
+function intoAmplifyErrorOrThrow(error: unknown): AmplifyError {
+    if (isAmplifyError(error)) {
+        return error
+    } else {
+        throw error
+    }
+}
+
+// =================
+// === AuthError ===
+// =================
+
+/** Object returned by the AWS Amplify library when an auth error occurs. */
+interface AuthError {
+    name: string
+    log: string
+}
+
+/** Hint to TypeScript if we can safely cast an `unknown` error to an `AuthError`. */
+function isAuthError(error: unknown): error is AuthError {
+    if (error != null && typeof error === 'object') {
+        return 'name' in error && 'log' in error
+    } else {
+        return false
+    }
+}
```

Comment:
Looks like a lot of duplicate code here (and throughout the rest of the file) from the non-mock definition of the class. If this is a mock, shouldn't it re-use the same types/interfaces and only change the underlying `impl`s?

Reply:
good point. i didn't think of that due to how the mock files are injected (?) into the app, but it *is* a simple change to import things from the original files